examples/lighting-app/k32w/main/AppTask.cpp

```diff
@@ -241,6 +241,12 @@ void AppTask::ButtonEventHandler(uint8_t pin_no, uint8_t button_action)
     else if (pin_no == BLE_BUTTON)
     {
         button_event.Handler = BleHandler;
+#if !(defined OM15082)
```

Comment:
should this not be using `defined(OM15082)` instead of `(defined OM15082)` ? I find the bracketing a bit odd ... gcc `defined` docs seem to only describe the brackets after the keyword.

Is this a style that is valid?

Reply:
I just tested it and it does work ... but feels unusual. Are there docs on why it works?

Reply:
Hey @andy31415, that syntax is basically the one described from GCC standard `defined OM15082` wrapped around parenthesis. Check out the reference [link](https://en.cppreference.com/w/cpp/preprocessor/conditional) under examples and also the godbolt that I [baked](https://godbolt.org/#z:OYLghAFBqd5QCxAYwPYBMCmBRdBLAF1QCcAaPECAM1QDsCBlZAQwBtMQBGAFlJvoCqAZ0wAFAB4gA5AAYppAFZdSrZrVAB9LclIj2yAnjqVMtdAGFUrAK4BbWiABMpU%2BgAyeWpgBydgEaYxE68AA6oQoRGtJY29k6kYRGGdB5evrYBQY68epgGUQwEzMQEMXYOzrn5dIXFBKk%2B/oHBukUlZXGVbfWejRnN2QCUuqjWxMgcUgD0UwDUACoAniGYs4ujxLNoWLMIgZiksySzrKjM6LNqs5jizLYh7AB0AKQyAILPjgDMnsg2O88vuYhAR0Kw8H5HghAdhXh9vr9/qtAcDQUYoTC4VjvlgqL1Zm8AELmOEzWYCWhoWy2UwEXb7UlzT5fXH4gAi2AAYtiWZg8V5ZgANACaAC0se9mXgqLMwLBWV4LkTzINZp8AGwa2YQBWYC4czmDVU8wLEY6fRwACUwi1mBD2xEwFpNZmlEvhPxlcp1fN6%2Bq5gxNxDNmwtBtmtFQdN16EOMdm4aEi3oN2dku%2BrjdkvTntlkBjEANRqDIbVjkc4cj0d9irjNb1hYDsyT9GY4jTHszVHdnjptmYnggxoA7IS4bMJ7NHQQxrRZjJAWPJcO2VJhqxpABWeQOWTyVDSczNjYTMtfTjyAjSORG0gAaxAXy%2Bjyfr7f7/VKmk3B31/30nkIQQBkUgrz3YY4FgJAqRCPB2DICgIBguDmmQYBOBkL4%2BDgghAiAiA/D/Ug/E8YpFmkC9SCpGl6AAeVoVhyL3UgsH7dR2CI/BHXyAA3TAgOYm48msXCiN7TAN2Y8E/GIMjLCwCjL2IPBbEU4Z%2BEYFgOJ4Pg6AIYQxEkZilE4FQ1E0bQVAhIDIGGVAQmSWgBMAiS8kckwzA6BxTNcBp0kyZREkiOgvMC8JgtoPymiCUyqkc2p2iscplDigpuii/oYtaOpQti9Len85pOGGIQTw4YqvykbdSF3ORSAPKRxAADnVABadVuFmYBkGQWYMJfbVcEIc1vlM2ZLHuFDQ1G1VzEvP9bwfJ8X3fVan0/SSfxqoiGsA4DQIW0hIMQEBRgIEIRPIShkPg5Q9WGrKNKYNhyt4DSDIkIiAHdZJCNTKuq2r/ykZlOFmL7CAQWZmrajqup6vqZBfebwOGJbnzWtbKq2oH6oA3R9rAm912kRxf2Y3aDtR0g%2BOICJjG4IA%3D%3D).